docs/decisions/0007-compile-common-theme-for-all-mfes.rst

```diff
@@ -0,0 +1,116 @@
+######################################
+
+
+Status
+******
+
+Draft
+
+Context
+*******
+
+Currently each Open edX MFE includes a small snippet of SCSS code that is common
+to all of them, which is the following four lines:
+
+.. code-block:: SCSS
+
+	@import "~@edx/brand/paragon/fonts";
+	@import "~@edx/brand/paragon/variables";
+	@import "~@edx/paragon/scss/core/core";
+	@import "~@edx/brand/paragon/overrides";
+
+These four lines, import the Paragon and bootstrap code and apply
+branding-specific overrides on top. They contain the bulk of the SCSS code
+generated for MFEs. Following this are usually imports for the frontend header
+and footer components, imports from the MFE own components and MFE-specific
+SCSS.
+
+This effectively means that all MFEs share a large chunk of common styling code
+with a comparatively much smaller set of styles applied over that.
+
+There is scope here to properly split the MFE stylesheet into multiple parts,
+one being the common code as represented by the above imports, and another for
+MFE-specific code, and build them separately.
+
+One thing that prevents this is that MFEs need these SCSS imports to get
+variables. In a number of cases these can be replaced with either
+Paragon/Bootstrap classes, or CSS variables. A separate ADR in `Paragon
+<https://github.com/openedx/paragon/pull/1388>`_ covers the situations where
+this isn't possible yet.
+
+Decision
+********
+
+Common cores styling that is to be applied to all MFEs should be part of a
+separate stylesheet and built separately from the MFE-specific stylesheet.
+
+Both these stylesheets will be included in each MFE, however the common
+stylesheet can be loaded from a common source for all MFEs.This will allowing
+for deploying the common theme stylesheet without redeploying the MFEs.
+
+The mechanism for loading this common stylesheet will need to be different from
+the mechanism for loading the MFE stylesheet. A JavaScript theme loader will
+contain the location of the SCSS file and will add this stylesheet to the
+document.
+
+The common theme package will be similar to the branding package, but can be
+deployed independently of MFEs. It will produce an CSS stylesheet, and a
+JavaScript loader that will load this stylesheet. When deploying MFEs, they can
+be provided a configuration variable that contains the location where this
+theme loader is deployed.
+
+The frontend-platform repo can host code that simplifies using themes. It can
+check if a theme loader is configured, and if so, dynamically load the theme
+from there. Otherwise, it can continue using the existing mechanism of building
+and deploying the theme for each MFE for backwards-compatibility. It can also
+provide hooks for loading the theme and potentially/eventually also selecting
+a theme.
+
+Consequences
+************
+
+With a single common stylesheet for all MFE, they should load quicker since the
+stylesheet can be cached once for all MFEs. It also makes theme rebuilds much
```

Comment:
I understand the theming benefits listed in the second half of this paragraph.
What I'm wondering about is the first benefit, loading the MFEs quicker since the stylesheet is cached. Could this also be achieved just by deploying all MFE's CSS assets to the same CDN folder? Because if those four lines (fonts, variables, core, overrides) are the same for each MFE, and the same version of Paragon and theme are used by each MFE, and webpack is configured correctly, it should produce a CSS chunk file for those core styles with a hash in its name, and that hash should be the same for each such MFE. So by deploying the CSS to the same CDN location, the different MFEs would still load the same common CSS file without any other changes. Or no?

Reply:
I think this will require making specific assumptions about how it's built and loaded from a CDN. Additionally, as mentioned elsewhere, currently different MFEs can use a slightly different version of Paragon, which will cause the hashes to not line up.